File: src/main/java/net/reflact/client/ClientData.java
package net.reflact.client;

import java.util.HashMap;
import java.util.Map;

public class ClientData {
    public static double maxMana = 100.0;
    public static double currentMana = 100.0;
    public static double maxHealth = 20.0;
    public static double currentHealth = 20.0;

    public static final Map<String, Long> spellCooldowns = new HashMap<>();
    public static final Map<String, Long> spellLastCast = new HashMap<>();

    public static class Waypoint {
        public String name;
        public int x, y, z;
        public int color;
        
        public Waypoint(String name, int x, int y, int z, int color) {
            this.name = name;
            this.x = x;
            this.y = y;
            this.z = z;
            this.color = color;
        }
    }
    
    public static final java.util.List<Waypoint> waypoints = new java.util.ArrayList<>();

    static {
        spellCooldowns.put("fireball", 5000L); // 5 seconds
        spellCooldowns.put("heal", 12000L); // 12 seconds
        
        waypoints.add(new Waypoint("Spawn", 0, 100, 0, 0xFF00FF00));
    }
}

File: src/main/kotlin/net/reflact/client/ClientData.kt
package net.reflact.client

import java.util.HashMap

object ClientData {
    var maxMana = 100.0
    var currentMana = 100.0
    var maxHealth = 20.0
    var currentHealth = 20.0

    val spellCooldowns: MutableMap<String, Long> = HashMap()
    val spellLastCast: MutableMap<String, Long> = HashMap()

    class Waypoint(var name: String, var x: Int, var y: Int, var z: Int, var color: Int)

    val waypoints: MutableList<Waypoint> = ArrayList()

    init {
        spellCooldowns["fireball"] = 5000L // 5 seconds
        spellCooldowns["heal"] = 12000L // 12 seconds

        waypoints.add(Waypoint("Spawn", 0, 100, 0, 0xFF00FF00.toInt()))
    }
}
